src/compiler.ts

```diff
@@ -7756,6 +7756,13 @@ export class Compiler extends DiagnosticEmitter {
           uniqueMap<string,Type>(flow.contextualTypeArguments)
         );
         if (!functionInstance || !this.compileFunction(functionInstance)) return module.unreachable();
+        if (functionInstance.hasDecorator(DecoratorFlags.BUILTIN)) {
+          this.error(
+            DiagnosticCode.Not_implemented_0,
+            expression.range, "First-class built-ins"
+          );
```

Comment:
Since `functionInstance` is known here, setting `this.currentType` to the function's type anyhow could prevent further compile errors that only come up due to this one.